content/en/integrations/guide/service-account-impersonation.md

```diff
@@ -0,0 +1,111 @@
+
+<div class="alert alert-warning">
+This feature is in beta. Log archiving to Google projects is not supported with service account impersonation. Errors and warnings about issues with service account impersonation are not fully represented in Datadog.
+</div>
+
+### Prerequisites
+
+* If your organization restricts identities by domain, you must add Datadog’s customer identity as an allowed value in your policy. Datadog’s customer identity: `C0147pk0i`
+
+* Service account impersonation and automatic project discovery rely on having certain roles and APIs enabled to monitor projects. Before you start, ensure the following APIs are enabled for the projects you want to monitor:
+  * [Cloud Resource Manager API][3]
+  * [Google Cloud Billing API][4]
+  * [Cloud Monitoring API][5]
+  * [Compute Engine API][6]
+  * [Cloud Asset API][7]
+  * [IAM API][8]
+
+### Setup
+
+### 1. Create your Google Cloud service account
+
+1. Open your [Google Cloud console][9].
+2. Navigate to **IAM & Admin** > **Service Accounts**.
+3. Click on **Create service account** at the top.
+4. Give the service account a unique name, then click **Create and continue**.
+5. Add the following roles to the service account:
+   * Monitoring Viewer
+   * Compute Viewer
+   * Cloud Asset Viewer
+   * Browser
+6. Click **Continue**, then **Done** to complete creating the service account.
+
+{{< img src="integrations/guide/service_account_impersonation/create-service-account.png" alt="Google Cloud console interface, showing the 'Create service account' flow. Under 'Grant this service account access to project', the four roles in the instructions are added." style="width:70%;">}}
+
+### 2. Add the Datadog principal to your service account
+
+1. In Datadog, navigate to the [**Integrations** > **Google Cloud Platform**][10].
+2. Click on **Add GCP Account**. If you have no configured projects, you are automatically redirected to this page.
+3. If you have not generated a Datadog principal for your org, click the **Generate Principal** button.
+4. Copy your Datadog principal and keep it for the next section.
+   {{< img src="integrations/guide/service_account_impersonation/datadog-principal.png" alt="Datadog interface, showing the 'Add New GCP Account' flow. The first step, 'Add Datadog Principal to Google,' features a text box where a user can generate a Datadog Principal and copy it to their clipboard. The second step, 'Add Service Account Email,' features a text box that the user can complete in section 3." style="width:70%;">}}
```

Comment:
Let's use an official screenshot instead. cc @dangteralyn can you please help here?

Reply:
**Is a gif acceptable/appropriate?**

![gcp-generate-principal](https://github.com/DataDog/documentation/assets/92749080/80b1e23e-c274-45ca-a5c5-ed2e14008e5c)

**Otherwise, here's still:** 

<img width="1195" alt="image" src="https://github.com/DataDog/documentation/assets/92749080/de37d31f-3328-4e3d-ae74-474be4e5892f">

Reply:
I think gifs are great if the motion will render well. @cswatt have we used gifs before?

Reply:
We typically prefer to use videos because of accessibility, and only in cases where it really enhances understanding to see the full thing. To be honest, I don't think clicking a button is really necessary to show in video/gif form.

Reply:
The format of the principal email address is outdated. A correctly formatted principal email would look like:

 `ddgci-b5ee8760a0ff148b3056@datadog-gci-sts-us1-prod.iam.gserviceaccount.com`

Reply:
@dangteralyn is this something we can fix? ^

Reply:
![image](https://github.com/DataDog/documentation/assets/92749080/6e8649df-0c09-44eb-aef1-44c05833d7bd)